packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -1,14 +1,13 @@
 import type { AnyClient } from "@gadgetinc/api-client-core";
 import { Provider as GadgetUrqlProvider, useQuery } from "@gadgetinc/react";
-import type { History, LocationOrHref } from "@shopify/app-bridge-react";
-import AppBridgeReact from "@shopify/app-bridge-react";
-import { AppBridgeContext } from "@shopify/app-bridge-react/context.js";
-import { getSessionToken } from "@shopify/app-bridge-utils";
-import { Redirect } from "@shopify/app-bridge/actions/index.js";
-import React, { memo, useContext, useEffect, useMemo, useState } from "react";
+import { useAppBridge } from "@shopify/app-bridge-react";
+import React, { memo, useEffect, useMemo, useState } from "react";
 import type { GadgetAuthContextValue } from "./index.js";
 import { GadgetAuthContext } from "./index.js";
 
+// if we haven't properly set up the shopify global then skip anything that requires app bridge
+const shopifyGlobalDefined = !!(globalThis && globalThis.shopify);
```

Comment:
we don't inject the shopify script tag if you don't have an api key or `@shopify/app-bridge-react` v4 in gadget. this check avoids trying to use the `useAppBridge` hook which throws whenever the shopify global hasn't been set up properly